tests/data/test_study/dataset_ingest_config.yml

```diff
@@ -25,6 +25,10 @@ extract_config_dir: 'extract_configs'
 
 transform_function_path: 'transform_module.py'
 
+expected_counts:
```

Comment:
Complicated. I do think that using unique_key is the right approach. Can we guarantee its availability after extract? I'm not sure but I think not, and I'm thinking specifically about specimens vs aliquots here.

What if we accept both but translate lone concepts as meaning unique_key and then happening only after transform?

Reply:
> I do think that using unique_key is the right approach in the general case.

What did you mean by general case? Unique keys don't get created until during transform stage, so may not be available for counting tests in post extract accounting. Unless...someone maps directly to the UNIQUE_KEY attribute in their extract config - which is allowed/supported (allows someone to override how the keys are created in case they don't like the default generation)

 > What if we accept both concepts and attributes but automatically translate lone concepts as meaning id (after extract) / unique_key (after transform)?

👍 I think this would satisfy everything we need